hail/python/hail/methods/impex.py

```diff
@@ -423,8 +423,23 @@ def export_vcf(dataset, output, append_to_header=None, parallel=None, metadata=N
         dictionary should be structured.
 
     """
-
     require_row_key_variant(dataset, 'export_vcf')
+    row_fields_used = {'rsid', 'info', 'filters', 'qual'}
+
+    fields_dropped = []
+    for f in dataset.globals:
+        fields_dropped.append((f, 'global'))
+    for f in dataset.col_value:
+        fields_dropped.append((f, 'column'))
+    for f in dataset.row_value:
+        if not f in row_fields_used:
+            fields_dropped.append((f, 'row'))
+
+    if fields_dropped:
+        ignored_str = ''.join(f'\n    {f!r} ({axis})' for f, axis in fields_dropped)
+        hl.utils.java.warn('export_vcf: ignored the following fields:' + ignored_str)
+        dataset = dataset.drop(*(f for f, _ in fields_dropped))
```

Comment:
Do we need to drop the fields here? I thought they just get ignored anyways.

Reply:
we don't need to, but dropping them here means we can prune them! For instance, in current master, if you do `sample_qc` > `export_vcf`, the sampleqc fields will be computed and then ignored. Now they won't be computed at all.